dask/cli.py

```diff
@@ -75,6 +77,61 @@ def config_get(key=None):
             exit(1)
 
 
+@config.command(name="set")
+@click.argument("key", default=None, required=False)
+@click.argument("value", default=None, required=False)
+def config_set(key=None, value=None):
+    """Set a Dask config key to a new value"""
+    if key in (None, ""):
+        click.echo(
+            click.style(
+                """Config key not specified. Are you looking for "dask config list"?"""
+            ),
+            err=True,
+        )
+        exit(1)
+    else:
+        try:
+            value = ast.literal_eval(value)
+        except (ValueError, SyntaxError):  # Not a literal
+            pass
```

Comment:
This is different from the same code in dask/config.py and doesn't process null/none correctly:
https://github.com/dask/dask/blob/9a1d4f14e29f279cd1eb3154367cc0806fcebbe4/dask/config.py#L260-L266

Maybe a minor refactor of `dask.config.collect_env` to parse the value from here would be more robust in the long term than trying to keep the functionality aligned in the two places?